### PR TITLE
[KernelGen][Nvidia] Add __lshift__ operator with Triton kernel

### PR DESCRIPTION
**Pull request by bwbwzzz**
## Summary
Adds a magic method wrapper for `__lshift__` (`<<` operator), which delegates to `bitwise_left_shift`. This enables the `<<` operator syntax for tensors.

## Testing
- Parametrized tests over multiple shapes and int16/int32 dtypes (12 cases)
- Validated against PyTorch `bitwise_left_shift` reference on device side
- Tested on: Nvidia H20

## Repository / Branch
- Target repository: `flagos-ai/FlagGems`
- Base branch: `master`

## Performance
Test command: `pytest benchmark/test_lshift__.py --level core` (NVIDIA H20)

### int16
| Size Detail | Torch Latency (ms) | Gems Latency (ms) | Speedup |
|---|---|---|---|
| 1073741824 | 1.609 | 1.511 | 1.065 |
| 64, 64 | 0.006 | 0.006 | 0.989 |
| 4096, 4096 | 0.031 | 0.031 | 1.008 |
| 64, 512, 512 | 0.031 | 0.031 | 0.999 |
| 1024, 1024, 1024 | 1.663 | 1.617 | 1.029 |

### int32
| Size Detail | Torch Latency (ms) | Gems Latency (ms) | Speedup |
|---|---|---|---|
| 1073741824 | 3.052 | 2.941 | 1.038 |
| 64, 64 | 0.006 | 0.006 | 0.972 |
| 4096, 4096 | 0.054 | 0.053 | 1.014 |
| 64, 512, 512 | 0.054 | 0.054 | 1.007 |
| 1024, 1024, 1024 | 3.054 | 2.943 | 1.038 |

**Arithmetic Mean Speedup: 1.016x** (10 cases, all PASS)

## Multi-backend Testing
| Backend | Accuracy Test | Benchmark | Speedup (mean) | Notes |
|---|---|---|---|---|
| Nvidia (H20) | PASS | PASS (10 cases) | 1.016 | Primary |
| Tianshu | PASS | PASS (30 cases) | 1.003 | — |
| Muxi | PASS | PASS (30 cases) | 1.003 | — |
| Ascend | PASS | — | — | — |
| Hygon | PASS | PASS (30 cases) | 0.991 | — |

## Files Changed
- `src/flag_gems/ops/lshift__.py`: Magic method wrapper delegating to bitwise_left_shift
- `tests/test_lshift__.py`: Accuracy test (12 cases, int16/int32)
- `benchmark/test_lshift__.py`: Performance benchmark
- `src/flag_gems/ops/__init__.py`: Register import and `__all__`
- `src/flag_gems/__init__.py`: Register to `_FULL_CONFIG`
- `conf/operators.yaml`: Add operator entry

### CODE REVIEW

File: benchmark/test_lshift.py [hunk 1]
```diff
@@ -0,0 +1,14 @@
+import pytest
+import torch
+
+from . import base, consts
+
+
+@pytest.mark.lshift__
```
Comment (bwbwzzz) [inline benchmark/test_lshift.py]:
```suggestion
@pytest.mark.lshift
```

File: benchmark/test_lshift.py [hunk 2]
```diff
@@ -0,0 +1,14 @@
+import pytest
+import torch
+
+from . import base, consts
+
+
+@pytest.mark.lshift
+def test_lshift():
+    bench = base.BinaryPointwiseBenchmark(
+        op_name="lshift__",
```
Comment (bwbwzzz) [inline benchmark/test_lshift.py]:
```suggestion
        op_name="lshift",
```

File: conf/operators.yaml [hunk 3]
```diff
@@ -4755,33 +4616,31 @@ ops:
       - Math
     stages:
       - alpha: '5.3'
-  - id: lt
-    description: Computes that `input` is less than `other` element-wise.
+  - id: lshift__
```
Comment (bwbwzzz) [inline conf/operators.yaml]:
```suggestion
  - id: lshift
```

File: src/flag_gems/__init__.py [hunk 4]
```diff
@@ -46,11 +41,10 @@ def torch_ge(v):
 
 
 _FULL_CONFIG = (
-    ("__and__.Scalar", bitwise_and_scalar),
-    ("__and__.Tensor", bitwise_and_tensor),
     ("__ilshift__.Tensor", __ilshift__),
     ("__ior__.Scalar", bitwise_or_scalar_),
     ("__ior__.Tensor", bitwise_or_tensor_),
+    ("__lshift__", __lshift__),
```
Comment (bwbwzzz) [inline src/flag_gems/__init__.py]:
```suggestion
    ("__lshift__.Tensor", __lshift__),
```

File: tests/test_lshift.py [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+    inp2 = torch.randint(low=0, high=8, size=shape, dtype=dtype, device="cpu").to(
+        flag_gems.device
+    )
+    ref_inp1 = utils.to_reference(inp1)
+    ref_inp2 = utils.to_reference(inp2)
+
+    # Reference: using bitwise_left_shift which is equivalent to __lshift__
+    ref_out = torch.bitwise_left_shift(ref_inp1, ref_inp2)
+    with flag_gems.use_gems():
+        res_out = torch.ops.aten.__lshift__(inp1, inp2)
```
Comment (bwbwzzz) [inline tests/test_lshift.py]:
```suggestion
        inp1 << inp2
```

File: tests/test_lshift.py [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+    inp2 = torch.randint(low=0, high=8, size=shape, dtype=dtype, device="cpu").to(
+        flag_gems.device
+    )
+    ref_inp1 = utils.to_reference(inp1)
+    ref_inp2 = utils.to_reference(inp2)
+
+    # Reference: using bitwise_left_shift which is equivalent to __lshift__
+    ref_out = torch.bitwise_left_shift(ref_inp1, ref_inp2)
+    with flag_gems.use_gems():
+        inp1 << inp2
```
Comment (bwbwzzz) [inline tests/test_lshift.py]:
```suggestion
        res_out = inp1 << inp2
```

File: tests/test_lshift.py [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+        flag_gems.device
+    )
+    inp2 = torch.randint(low=0, high=8, size=shape, dtype=dtype, device="cpu").to(
+        flag_gems.device
+    )
+    ref_inp1 = utils.to_reference(inp1)
+    ref_inp2 = utils.to_reference(inp2)
+
+    # Reference: using bitwise_left_shift which is equivalent to __lshift__
+    ref_out = torch.bitwise_left_shift(ref_inp1, ref_inp2)
```
Comment (bwbwzzz) [inline tests/test_lshift.py]:
```suggestion
    ref_out = 、ref_inp1 << ref_inp2
```

File: tests/test_lshift.py [hunk 8]
```diff
@@ -0,0 +1,27 @@
...
+        flag_gems.device
+    )
+    inp2 = torch.randint(low=0, high=8, size=shape, dtype=dtype, device="cpu").to(
+        flag_gems.device
+    )
+    ref_inp1 = utils.to_reference(inp1)
+    ref_inp2 = utils.to_reference(inp2)
+
+    # Reference: using bitwise_left_shift which is equivalent to __lshift__
+    ref_out = 、ref_inp1 << ref_inp2
```
Comment (bwbwzzz) [inline tests/test_lshift.py]:
```suggestion
    ref_out = ref_inp1 << ref_inp2
```

File: src/flag_gems/ops/lshift.py [hunk 9]
```diff
@@ -0,0 +1,22 @@
+# Generated by KernelGen: https://github.com/flagos-ai/KernelGen
+import logging
+
+import flag_gems  # noqa: F401
+from flag_gems.ops.bitwise_left_shift import bitwise_left_shift
```
Comment (bwbwzzz) [inline src/flag_gems/ops/lshift.py]:
```suggestion
# Generated by KernelGen: https://github.com/flagos-ai/KernelGen
import logging

import flag_gems  # noqa: F401
from flag_gems.utils import pointwise_dynamic
imort torch
```
